numcodecs/bz2.py

```diff
@@ -28,29 +27,21 @@ def __init__(self, level=1):
 
     def encode(self, buf):
 
-        # deal with lack of buffer support for datetime64 and timedelta64
-        buf = handle_datetime(buf)
-
         if isinstance(buf, np.ndarray):
 
             # cannot compress object array
             if buf.dtype == object:
                 raise ValueError('cannot encode object array')
 
-            # if numpy array, can only handle C contiguous directly
-            if not buf.flags.c_contiguous:
-                buf = buf.tobytes(order='A')
+        buf = to_buffer(buf)
```

Comment:
Love how this is simplifying the codec implementations!

Reply:
Definitely. Wonder if it's worth extending this to the Cython implementations?

Reply:
IIRC the Cython implementations should already be using buffers in both Py2 and Py3, there is some C-level compatibility code in compat_ext.pyx. But by all means take a look, I may well have missed something!

Reply:
Did look briefly. Might save that for another PR though.